src/main/java/org/gridsuite/modification/server/modifications/BatteryModification.java

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.modification.server.modifications;
+
+import com.powsybl.commons.reporter.Report;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.TypedValue;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.ActivePowerControl;
+import com.powsybl.iidm.network.extensions.ActivePowerControlAdder;
+import com.powsybl.network.store.iidm.impl.MinMaxReactiveLimitsImpl;
+import org.gridsuite.modification.server.NetworkModificationException;
+import org.gridsuite.modification.server.dto.BatteryModificationInfos;
+import org.gridsuite.modification.server.dto.ReactiveCapabilityCurveModificationInfos;
+import org.springframework.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static org.gridsuite.modification.server.NetworkModificationException.Type.MODIFY_BATTERY_ERROR;
+/**
+ * @author Ghazwa Rehili <ghazwa.rehili at rte-france.com>
+ */
+public class BatteryModification extends AbstractModification {
+
+    private final BatteryModificationInfos modificationInfos;
+
+    private static final String LIMITS = "Limits";
+    private static final String ACTIVE_LIMITS = "Active limits";
+    private static final String SETPOINTS = "Setpoints";
+    public static final String REACTIVE_LIMITS = "Reactive limits";
+
+    public BatteryModification(BatteryModificationInfos modificationInfos) {
+        this.modificationInfos = modificationInfos;
+    }
+
+    private static NetworkModificationException makeBatteryException(String batteryId, String msgSuffix) {
+        return new NetworkModificationException(MODIFY_BATTERY_ERROR, "Battery '" + batteryId + "' : " + msgSuffix);
+    }
+
+    @Override
+    public void check(Network network) throws NetworkModificationException {
+        if (network.getBattery(modificationInfos.getEquipmentId()) != null) {
```

Comment:
Why not throwing an exception if battery is not found ?

Reply:
done